### PR TITLE
Added explosive beartraps.

### PR DESCRIPTION
**Pull request by VelardAmakar**
Attach an IED to a beartrap to blow someone's legs off when it clamps.
Use a screwdriver to get the bomb off the trap.

### CODE REVIEW

**Comment by CptWad** [comment]
Hell yeah EXPLODING FEET :+1:

**Comment by d3athrow** [comment]
But muh wet floor sign mines

File: code/defines/obj/weapon.dm [hunk 1]
```diff
@@ -416,8 +417,46 @@
 		icon_state = "beartrap[armed]"
 		user << "<span class='notice'>[src] is now [armed ? "armed" : "disarmed"]</span>"
 
+/obj/item/weapon/legcuffs/beartrap/attackby(var/obj/item/I, mob/user as mob) //Let's get explosive.
+	if(istype(I, /obj/item/weapon/grenade/iedcasing))
```
Comment (ComicIronic) [inline code/defines/obj/weapon.dm]: No check for already having a bomb?

File: code/game/objects/items/weapons/grenades/ghettobomb.dm [hunk 2]
```diff
@@ -84,6 +84,18 @@
...
+		if(istype(boomtrap.loc, /mob/living/carbon/human))
+			var/mob/living/carbon/human/H = loc.loc
+			if(H.legcuffed)
+				var/datum/organ/external/l = H.get_organ("l_leg")
+				var/datum/organ/external/r = H.get_organ("r_leg")
+				if(l && !(l.status & ORGAN_DESTROYED))
+					l.status |= ORGAN_DESTROYED
+				if(r && !(r.status & ORGAN_DESTROYED))
+					r.status |= ORGAN_DESTROYED
+				del(H.legcuffed)
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/grenades/ghettobomb.dm]: qdel

File: code/defines/obj/weapon.dm [hunk 3]
```diff
@@ -416,8 +417,46 @@
...
 /obj/item/weapon/legcuffs/beartrap/Crossed(AM as mob|obj)
 	if(armed)
+		if(IED && isturf(src.loc))
+			IED.active = 1
+			IED.overlays -= image('icons/obj/grenade.dmi', icon_state = "improvised_grenade_filled")
+			IED.icon_state = initial(icon_state) + "_active"
+			IED.assembled = 3
+			var/turf/bombturf = get_turf(src)
+			var/area/A = get_area(bombturf)
+			var/log_str = "[key_name(usr)]<A HREF='?_src_=holder;adminmoreinfo=\ref[AM]'>?</A> has triggered an IED-rigged [name] at <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[bombturf.x];Y=[bombturf.y];Z=[bombturf.z]'>[A.name] (JMP)</a>."
```
Comment (ComicIronic) [inline code/defines/obj/weapon.dm]: I suggest you log priming it with a bomb as well.

**Comment by VelardAmakar** [comment]
>But muh wet floor sign mines

why not both
more boom is good
less boom is bad
more boom good for boom

File: code/defines/obj/weapon.dm [hunk 4]
```diff
@@ -416,8 +417,54 @@
...
+			user << "<span class='warning'>This beartrap already has an IED hooked up to it!</span>"
+			return
+		IED = I
+		switch(IED.assembled)
+			if(0,1) //if it's not fueled/hooked up
+				user << "<span class='warning'>You haven't prepared this IED yet!</span>"
+				IED = null
+				return
+			if(2,3)
+				user.drop_item()
```
Comment (ComicIronic) [inline code/defines/obj/weapon.dm]: Just compress this to drop_item(src), it's more code sane.

File: code/defines/obj/weapon.dm [hunk 5]
```diff
@@ -416,8 +417,54 @@
...
+				user << "<span class='danger'>You shouldn't be reading this message! Contact a coder or someone, something broke!</span>"
+				IED = null
+				return
+	if(istype(I, /obj/item/weapon/screwdriver))
+		if(IED)
+			IED.loc = get_turf(src.loc)
+			IED = null
+			user << "<span class='notice'>You remove the IED from the [src].</span>"
+			return
+		..()
```
Comment (ComicIronic) [inline code/defines/obj/weapon.dm]: Return ..(), please. This should also not be tabbed under the if.

File: code/defines/obj/weapon.dm [hunk 6]
```diff
@@ -427,6 +474,8 @@
 					src.loc = H
 					H.update_inv_legcuffed()
 					H << "\red <B>You step on \the [src]!</B>"
```
Comment (ComicIronic) [inline code/defines/obj/weapon.dm]: Could you fix this span as well?

File: code/game/objects/items/weapons/grenades/ghettobomb.dm [hunk 7]
```diff
@@ -84,6 +84,18 @@
 /obj/item/weapon/grenade/iedcasing/prime() //Blowing that can up
 	update_mob()
 	explosion(src.loc,-1,0,2)
+	if(istype(loc, /obj/item/weapon/legcuffs/beartrap))
+		var/obj/item/weapon/legcuffs/beartrap/boomtrap
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/grenades/ghettobomb.dm]: boomtrap = loc

**Comment by VelardAmakar** [comment]
hhhhhhhh that should be the last of it

**Pull request merged**